src/software/firmware/srcs/integration_test.cpp

```diff
@@ -0,0 +1,253 @@
+/******************************************************************************
+ * @author Makers For Life
+ * @copyright Copyright (c) 2020 Makers For Life
+ * @file integration_test.cpp
+ * @brief Entry point of integration test program
+ *****************************************************************************/
+
+#pragma once
+
+#include "../includes/config.h"
+#if MODE == MODE_INTEGRATION_TEST
+
+// INCLUDES ===================================================================
+
+// External
+#include <Arduino.h>
+#include <OneButton.h>
+
+// Internal
+#include "../includes/battery.h"
+#include "../includes/blower.h"
+#include "../includes/debug.h"
+#include "../includes/parameters.h"
+#include "../includes/pression.h"
+#include "../includes/pressure_valve.h"
+#include "../includes/screen.h"
+
+// State machine states
+#define STEP_WELCOME 0
+#define STEP_BLOWER_TEST 1
+#define STEP_VALVE_BLOWER_TEST 2
+#define STEP_VALVE_PATIENT_TEST 3
+#define STEP_VALVE_BLOWER_LEAK_TEST 4
+#define STEP_VALVE_PATIENT_LEAK_TEST 5
+#define STEP_PRESSURE_TEST 6
+#define STEP_BATTERY_TEST 7
+
+#define NUMBER_OF_STATES 8
+
+#define UNGREEDY(is_drawn, statement)                                                              \
+    if (is_drawn == 0) {                                                                           \
+        statement;                                                                                 \
+        is_drawn = 1;                                                                              \
+    }
+
+static uint8_t step = STEP_WELCOME;
+static uint8_t is_drawn = false;
+
+PressureValve servoBlower;
+PressureValve servoPatient;
+HardwareTimer* hardwareTimer1;
+HardwareTimer* hardwareTimer3;
+Blower* blower_pointer;
+Blower blower;
+
+int32_t last_time = millis();
+
+void changeStep(uint8_t new_step) {
+    step = new_step;
+    is_drawn = 0;
+}
+
+//! This function displays 2 lines of 20 characters (or less)
+void display(char line1[], char line2[]) {
+    resetScreen();
+    screen.setCursor(0, 0);
+    screen.print(line1);
+    screen.setCursor(0, 1);
+    screen.print(line2);
+}
+
+//! This function displays only a message on line
+void displayLine(char msg[], uint8_t line) {
+    screen.setCursor(0, line);
+    screen.print("                    ");
+    screen.setCursor(0, line);
+    screen.print(msg);
+}
```

Comment:
Should we merge this code between `integration_test.cpp` and `qualification.cpp`?